python/rpdk/java/resolver.py

```diff
@@ -2,7 +2,7 @@
 
 PRIMITIVE_TYPES = {
     "string": "String",
-    "integer": "Integer",
+    "integer": "Long",
```

Comment:
Is it backward compatible?  I'm afraid it will break existing resources have integer in their schema

Reply:
That's part of the reason why I opened it as a draft pr -- wanted to do some research to see if this was a safe change to make.

From looking it seems that if anyone is directly passing Integers from the pojo to an API request that requires an Integer, that would break. Might need a more sophisticated approach than just changing it here.

Reply:
Int32 constraints can be achieved with use of `minimum` and `maximum` in the schema. If the actual downstream API calls are expecting a Java `Integer` in their payload, then they'll get a compile error on next build and it's a signal they'll need to update something. 

This doesn't break anything already published, so we should make this change before it gets too hairy. Suggest we could also add tests around this to signal there's a change in the POJO, but I'm not sure what those would look like.

Reply:
Could we introduce something in the `.rpdk-config` that we can prefer to use Long instead of Integer?  It won't break existing customer even in the build time.   The java plugin only need to read the `.rpdk-config` for a config and decide to use Integer or Long, but by default it's Integer.

Reply:
FWIW I'd appreciate the solution proposed by @rjlohan, as it would allow us to distinguish between individual `Integer`s/`Long`s explicitly, unlike the other solutions which would require us to have *everything* `Long` (to be fair, having everything `Long` is still better, but would require us to do a conversion `Long`->`Integer` in quite a few places).

Reply:
If we go with using constraints as the determination for `int32` vs `long`/`int64` we probably just need to have a check on whether the `maximum` is `< 2147483647` (using whatever const Python3 has) to determine if we codegen as `int32`. Else we use `long`. Another alternative would be to introduce a new `long` type to our metaschema to be explicit.

Reply:
Yeah, I'm ok with determining this based on the `maximum`. If you want to go with introducing new type, I'd much rather see something along the line of OpenAPI spec, which has `type: integer` and it provides additional information in another property, e.g. `format: int32` or `format: int64`: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#dataTypeFormat

Reply:
The `format` approach is a better idea, let's do that.